internal/txsubmitter/solana/solana.go

```diff
@@ -73,22 +88,119 @@ func (s solanaTxSubmitter) Transfer(token *model.Token, to, amount string) (txHa
 		Signers: []types.Account{fromAcc},
 	})
 	if err != nil {
-		s.logger.Error(err, "[solana.Transfer] types.NewTransaction() failed")
+		l.Error(err, "[transferNative] types.NewTransaction() failed")
 		return txHash, err
 	}
 
 	// send tx
 	txHash, err = s.client.SendTransaction(context.Background(), tx)
 	if err != nil {
-		s.logger.Error(err, "[solana.Transfer] client.SendTransaction() failed")
+		l.Error(err, "[transferNative] client.SendTransaction() failed")
 		return txHash, err
 	}
 
-	// add tx to pool
-	s.receivedTxCh <- fmt.Sprintf("%s-%s", txHash, token.ChainId)
 	return txHash, nil
 }
 
+func (s solanaTxSubmitter) transferSpl(token *model.Token, to, amount string) (txHash string, err error) {
+	l := s.logger.WithFields(logrus.Fields{"token": token.Symbol, "to": to, "amount": amount})
+	fromAcc, _ := types.AccountFromBase58(s.privKey)
+	fromPubKey := fromAcc.PublicKey
+
+	// to fetch recent blockhash
+	res, err := s.client.GetLatestBlockhash(context.Background())
+	if err != nil {
+		l.Error(err, "[transferSpl] client.GetLatestBlockhash() failed")
+		return txHash, err
```

Comment:
can return only, same for other similar cases. Cmiiw 
```suggestion
		return 
```